ironfish/src/wallet/account.ts

```diff
@@ -461,6 +461,19 @@ export class Account {
           hash: noteHash,
         })
       }
+
+      const decryptedNoteForSpender = note.decryptNoteForSpender(this.outgoingViewKey)
```

Comment:
Decrypting notes is slow, so we're trying to never do decryption on the main thread and to move it to worker tasks instead.

It should also be possible to look up a spent note for an account using the nullifier and the nullifierToNoteHash data store

Reply:
You're right, we only store notes that the account owns!

Sorry, I misunderstood the intent here and was thinking of including the notes that are _spent_ in a transaction.

It would be useful to have a way for Tom to verify that he sent a particular note to Amy. We do want to avoid decrypting in the main thread, though, especially for every note one at a time. For a transaction with many output notes, like a pool payout, this would take a very long time.

I think we'll want to either move this decryption to a worker task, or update the wallet to have some storage of output notes from the transactions that an account creates.

Reply:
Note decryption has moved to workerPool in the latest commit.